doc/DifferentiableManifoldRecipe.md

```diff
@@ -0,0 +1,180 @@
+# Differentiable Manifold Recipe
+
+This recipe describes how to express a differentiable manifold in Swift. The
+main payoffs of using this recipe are:
+
+ * You can choose arbitrary global and local coordinate systems for your
+   manifolds.
+ * Swift's Automatic Differentiation works with the coordinate system that you
+   specify. In particular, the tangent vectors that flow through the
+   automatically-generated derivatives are exactly the tangent vectors
+   corresponding to the local coordinate system you specify.
+
+Note: We want to make the recipe simpler. See
+[issue #45](https://github.com/borglab/SwiftFusion/issues/45).
+
+## Steps
+
+### 1. Choose a global coordinate system.
+
+Create a `struct` whose values represent points on the manifold. Every point on
+the manifold must be representable by at least one `struct` value. Every
+`struct` value must represent at most one point on the manifold.
```

Comment:
@marcrasi Doesn't this simply translate to: “Every point on the manifold must be representable by exactly one `struct` value?”  If so, why not say it that way?

Reply:
[I'm a little concerned about the idea that the values must cover the entire manifold (with no caveats), since obviously our numerical representations are limited.